S3/S3.py

```diff
@@ -407,17 +407,21 @@ def bucket_create(self, bucket, bucket_location = None, extra_headers = None):
             headers.update(extra_headers)
 
         body = ""
-        if bucket_location and bucket_location.strip().upper() != "US" and bucket_location.strip().lower() != "us-east-1":
-            bucket_location = bucket_location.strip()
-            if bucket_location.upper() == "EU":
-                bucket_location = bucket_location.upper()
-            body  = "<CreateBucketConfiguration><LocationConstraint>"
-            body += bucket_location
-            body += "</LocationConstraint></CreateBucketConfiguration>"
-            debug("bucket_location: " + body)
-            check_bucket_name(bucket, dns_strict = True)
+        if self.config.bucket_name_quirks:
+            # We are explicitly not AWS
+            check_bucket_name(bucket, dns_strict = False, name_quirks = True)
         else:
-            check_bucket_name(bucket, dns_strict = False)
+            if bucket_location:
```

Comment:
The CreateBucketConfiguration should be set in all cases, that we use "name quirks" or not.

For the moment, keep the existing code but remove the "check_bucket_name" of this block.
(In the future, I will change the EU/US thing to avoid some specific issues, but it can be done after).

After the previous block, add a new block of code:
```
if self.config.bucket_name_quirks:
    checkbucket(dns_strict = False)
else:
   check_bucket(dns_strict= True)
```